### PR TITLE
JSON loader

### PR DESCRIPTION
**Pull request by avsolatorio**
This implements a loader of text passages in JSON format. The `jq` syntax is used to define a schema for accessing the relevant contents from the JSON file. This requires dependency on the `jq` package: https://pypi.org/project/jq/.

### CODE REVIEW

**Review by dev2049 — COMMENTED** [review]
nice, thanks @avsolatorio for adding! could we add a brief demo notebook (see `docs/modules/indexes/document_loaders/examples`) and integration test (`tests/integration_tests/document_loaders`)?

File: langchain/document_loaders/json_loader.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+    def load(self) -> List[Document]:
+        """Load and return documents from the JSON file."""
+
+        body_text = self._jq_schema.input(json.loads(self.file_path.read_text())).all()
+        docs = []
+
+        for i, text in enumerate(body_text, 1):
+            metadata = dict(
+                source=self.file_path.as_posix(),
```
Comment (vowelparrot) [inline langchain/document_loaders/json_loader.py:45]: If it's as posix, how would that impact systems running on Windows? Any reason to choose that vs. `str(self.file_path)`?

Comment (hwchase17) [reply langchain/document_loaders/json_loader.py:45]: +1

File: langchain/document_loaders/json_loader.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        """Load and return documents from the JSON file."""
+
+        data = self._jq_schema.input(json.loads(self.file_path.read_text()))
+
+        # Perform some validation
+        # This is not a perfect validation, but it should catch most cases
+        # and prevent the user from getting a cryptic error later on.
+        if self._content_key is not None:
+            sample = data.first()
+            assert isinstance(sample, dict)
```
Comment (dev2049) [inline langchain/document_loaders/json_loader.py:57]: nit-picky point, but convention in package so far has been to more explicitly validate and raise errors (usually ValueErrors). so something like 
```python
if not isinstance(sample, dict):
  raise ValueError(f'expected foo bar but received baz: {sample}')
```

Comment (avsolatorio) [reply langchain/document_loaders/json_loader.py]: TIL. Updated!

File: langchain/document_loaders/json_loader.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            file_path (Union[str, Path]): The path to the JSON file.
+            jq_schema (str): The jq schema to use to extract the data or text from the JSON.
+            content_key (str): The key to use to extract the content from the JSON if the jq_schema results to a list of objects (dict).
+            metadata_func (Callable[Dict, Dict]): A function that takes in the JSON object extracted by the jq_schema and the default metadata and returns a dict of the updated metadata.
+        """
+        try:
+            import jq  # noqa:F401
+        except ImportError:
+            raise ValueError(
+                "jq package not found, please install it with " "`pipenv install jq`"
```
Comment (dev2049) [inline langchain/document_loaders/json_loader.py:39]: nit: combine strings and would just do pip (instead of pipenv, which is bit less universal)

**Review by dev2049 — APPROVED** [review]
one nit but otherwise lgtm, thanks @avsolatorio!!

File: langchain/document_loaders/json_loader.py (hunk 3 above)
Comment (avsolatorio) [reply langchain/document_loaders/json_loader.py]: Oh, right! I also had to fix some linting issues so I just included this change in the latest commit. Thanks!

**Pull request merged**